lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -725,11 +725,6 @@ LogicalResult ExpandDimsOp::canonicalize(ExpandDimsOp op,
   //    broadcast(expand_dims(broadcast))
   // -> broadcast(broadcast(expand_dims))
   // -> broadcast(expand_dims)
-  //
-  // Note we can't do this if x is a scalar, because expandDims requires a
-  // tensor input.  (We could change it to broadcast(expand_dims(splat(x))), but
-  // it's unclear that's better.)  broadcast currently accepts scalar inputs,
-  // but it probably shouldn't (just use splat instead).
   if (auto broadcast = dyn_cast<triton::BroadcastOp>(definingOp)) {
     auto src = broadcast.getSrc();
     if (auto srcTy = src.getType().dyn_cast<RankedTensorType>()) {
```

Comment:
You should be able to get rid of this check now (that's what the comment was referring to)

Reply:
Thanks, removed the check.

Reply:
> Verified that the only places that call create_broadcast appropriately call create_splat, so no changes are needed to semantic.py and friends.

Indeed - initially I made changes wrongfully in this file, but at least I remembered to check it later :smile: